lib/puppet/provider/nameservice.rb

```diff
@@ -261,6 +269,10 @@ def groups
     groups.join(",")
   end
 
+  def groups
+    @groups ||= fetch_groups
```

Comment:
See commit 0c8da753e0909720be84a3b423dd1eb0d8e402b4. The list property can end up querying the provider several times now.

I didn't want to add the caching to the list property, since other property types don't do caching, and caching in providers isn't all that unusual.